mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -852,6 +874,11 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
 
     auto paddedInputMemRefType =
         MemRefType::get(paddedInputShape, inputElementType);
+    // set domain and range attributes.
```

Comment:
Since every transformation will have a output_bounds and source_bounds (I think we should call it "lower_bound" and "upper_bound" instead to avoid confusion), the process of adding these attributes should be be done in an automatic way (like inside the constructor of a transformation in C++, or in "make_xxx_transform" function in MLIR), instead of do it manually after every instance of new transformation.

Reply:
@asroy ignore my earlier comment, i found some codes to copy from and am working on this feature. will submit a new commit later today.